src/Microsoft.Diagnostics.Monitoring.Options/AzureAdOptions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed partial class AzureAdOptions
+    {
+        private const string DefaultInstanceId = "https://login.microsoftonline.com";
+        private const string DefaultTenantId = "organizations";
+
+        [Display(
+            ResourceType = typeof(OptionsDisplayStrings),
+            Description = nameof(OptionsDisplayStrings.DisplayAttributeDescription_AzureAdOptions_Instance))]
+        [DefaultValue(DefaultInstanceId)]
+        public string Instance { get; set; } = DefaultInstanceId;
```

Comment:
Don't assign the property with a default value; this causes any test that is applying configuration via the object model to implicitly set these properties. Instead, add extension methods (e.g. `GetInstanceId(this AzureAdOptions)` and `GetTenantId(this AzureAdOptions)` to get the value of the property or fallback to the default value. Then use the methods instead of the properties in the rest of the code.

Reply:
Can you specifically point to some other options class that is used in binding that has a default value set? All of the struct primitive types should be nullable and the strings should not have a default value. Note that I'll talking about the default value assignment, not the `[DefaultValue(...)]` attribute, which is necessary to describe the default value in the schema.